### PR TITLE
KOGITO-6045 - Scroll automatically expandable canvas

### PR DESCRIPTION
**Pull request by Josephblt**
The canvas now expands/scrolls automatically anytime a primitive is dragged beyond control limits.

**JIRA**: [KOGITO-6045](https://issues.redhat.com/browse/KOGITO-6045)

**DMN Editor**: [Web App (.war)](https://drive.google.com/file/d/1VgcMBedpD5w6-S4j07e9U1XRO20DsI5w/view?usp=sharing)

**BPMN Editor**: [Web App (.war)](https://drive.google.com/file/d/1q4xiMKpMzjYOu1MTGJZ_sh4gyeVoZqmX/view?usp=sharing)

**VS Code**: [VSCode Extension](https://drive.google.com/file/d/1h4dLGBfL5Og5h8Gqf-SLH4Zn-WWeuGfe/view?usp=sharing)

### CODE REVIEW

**Comment by domhanak** [comment]
@Josephblt Hi Wagner, I did a quick check in VSCode and I encounter an issue where the mouse pointer gets moved past the editor frame. Would it be possible to make it stick to the node and not leave canvas frame until I release the mouse button?

**Comment by romartin** [comment]
@Josephblt let's try to move this PR foward please, a few days with no updates. as soon as we can close this, much better! it will get early into hte releases. Thanks in advance!

**Comment by domhanak** [comment]
I reported this minor issue going forward - https://issues.redhat.com/browse/KOGITO-6127

**Review by domhanak — APPROVED** [review]
LGTM. Tested in DMN and BPMN editors. Reported issue is still valid however it is marked as optional.
https://issues.redhat.com/browse/KOGITO-6127

**Comment by Josephblt** [comment]
Hello @domhanak, @romartin. I need you guys to check out this again. I made some changes and I believe this is way better. 
I'm going to add some links with videos to show the main differences so you know what to test.

There are 3 main changes:
- **Scroll Bar live update when Panning** - When panning (alt + mouse move) the scroll bar updates as you go. Pay attention to the Scroll Bar to see differences.
Pan Scroll Bar Demo (Old Version)
Pan Scroll Bar Demo (New Version)

- **Scroll Bar live update when Dragging** - When dragging any node/point the scroll bar updates as you go.
Drag Scroll Bar Demo (Old Version)
Drag Scroll Bar Demo (New Version)

- **Prevent Viewport jumps after Drag** - After dragging a node/point the Viewport would recalculate its limits and "jump" attempting to fit it's contents correctly.
Viewport Scroll Bar Demo (Old Version)
Viewport Scroll Bar Demo (New Version)

Any questions guys please fell free to ask. Thanks.

**Comment by domhanak** [comment]
@Josephblt thanks, will retest again from scratch once there is a review from @romartin

File: lienzo-core/src/main/java/com/ait/lienzo/client/widget/panel/impl/LienzoPanelHandlerManager.java [hunk 1]
```diff
@@ -92,11 +98,23 @@
...
+    private int lastDragMoveY;
+
+    private MouseEvent lastDragMouseEvent;
+
+    private TouchEvent lastDragTouchEvent;
+
     private IPrimitive<?> m_drag_node = null;
 
-    private IPrimitive<?> m_over_prim = null;
+    private Timer dragLimitsTimer;
```
Comment (romartin) [inline lienzo-core/src/main/java/com/ait/lienzo/client/widget/panel/impl/LienzoPanelHandlerManager.java:115]: I see the `Timer` being used is from GWT. We just did a big work recently to drop as much gwt stuff as possible. So the idea is to try not to add more GWT stuff, if possible. You think could you please change it to j2cl? (`DomGlobal.setTimeOut`, etc). Thanks!

**Review by romartin — COMMENTED** [review]
Hey @Josephblt 

Thanks for this work. looks good! There are several changes in lienzo core, which is not easy to manage. So thanks for this.

I just added a minor review comment on the code. If you can refactor it, good, otherwise I don't think it will block the PR, but wouod be nice not using GWT abymore, at tleast in Lienzo.

**Review by romartin — APPROVED** [review]

**Review by domhanak — APPROVED** [review]
LGTM.

**Pull request merged**